src/checker/http_checker.rs

```diff
@@ -119,11 +119,17 @@ fn connection_error(err: &reqwest::Error) -> Option<String> {
     None
 }
 
-fn hyper_error(err: &reqwest::Error) -> Option<String> {
+fn hyper_error(err: &reqwest::Error) -> Option<(CheckStatusReasonType, String)> {
     let mut inner = &err as &dyn Error;
     while let Some(source) = inner.source() {
         if let Some(hyper_error) = source.downcast_ref::<hyper::Error>() {
-            return Some(hyper_error.to_string());
+            if hyper_error.is_incomplete_message() {
+                return Some((
+                    CheckStatusReasonType::ConnectionError,
+                    hyper_error.to_string(),
```

Comment:
Will this string end up being messy, similar to our other errors?

Reply:
no it won't be, this will be nice and human readable, as hypers implementation parses their errors.